File: src/main/java/org/learning/dlearning_backend/dto/request/UserProfileRequest.java
package org.learning.dlearning_backend.dto.request;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.*;
import lombok.experimental.FieldDefaults;
import org.learning.dlearning_backend.common.CourseLevel;
import org.learning.dlearning_backend.common.Gender;

import java.time.LocalDate;

@Setter
@Getter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
@JsonInclude(JsonInclude.Include.NON_NULL)
public class UserProfileRequest {

    String avatar;
    String firstName;
    String lastName;
    String phoneNumber;

    Gender gender;

    @JsonFormat(pattern = "yyyy-MM-dd")
    LocalDate dateOfBirth;

    String address;

    String description;
    CourseLevel courseLevel;

}

File: libs/shared/src/lib/models/request/user-profile.request.ts
import { CourseLevel } from '../enum/course-level.enum';

export interface UserProfileRequest {
  avatar: string;
  firstName: string;
  lastName: string;
  gender: string;
  phoneNumber: string;
  dateOfBirth: string;
  address: string;
  description: string;
  courseLevel: CourseLevel;
}
